### PR TITLE
[serve] rename http spcific classes and files to be generic

### LINKED ISSUE #38750: [Serve]  Rename `HTTPState`, `HTTPProxyActor`, and related code
### Description

There are quite a few class/ files named after http such as `HTTPState` and `HTTPProxyActor`. Let's rename them to be more generic since they are now handling both HTTP and gRPC proxies.

### Use case

_No response_

### PR DESCRIPTION
**Pull request by GeneDer**
## Why are these changes needed?

Rename all HTTP specific classes and files to be generic

## Related issue number

Closes: https://github.com/ray-project/ray/issues/38750

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/serve/advanced-guides/dev-workflow.md [hunk 1]
```diff
@@ -52,7 +52,7 @@ Now run the following command in your terminal:
 serve run local_dev:app
 # 2022-08-11 11:31:47,692 INFO scripts.py:294 -- Deploying from import path: "local_dev:app".
 # 2022-08-11 11:31:50,372 INFO worker.py:1481 -- Started a local Ray instance. View the dashboard at http://127.0.0.1:8265.
-# (ServeController pid=9865) INFO 2022-08-11 11:31:54,039 controller 9865 http_state.py:129 - Starting HTTP proxy with name 'SERVE_CONTROLLER_ACTOR:SERVE_PROXY_ACTOR-dff7dc5b97b4a11facaed746f02448224aa0c1fb651988ba7197e949' on node 'dff7dc5b97b4a11facaed746f02448224aa0c1fb651988ba7197e949' listening on '127.0.0.1:8000'
+# (ServeController pid=9865) INFO 2022-08-11 11:31:54,039 controller 9865 proxyproxy_state.py:129 - Starting HTTP proxy with name 'SERVE_CONTROLLER_ACTOR:SERVE_PROXY_ACTOR-dff7dc5b97b4a11facaed746f02448224aa0c1fb651988ba7197e949' on node 'dff7dc5b97b4a11facaed746f02448224aa0c1fb651988ba7197e949' listening on '127.0.0.1:8000'
```
Comment (zcin) [inline doc/source/serve/advanced-guides/dev-workflow.md:55]:
```suggestion
# (ServeController pid=9865) INFO 2022-08-11 11:31:54,039 controller 9865 proxy_state.py:129 - Starting HTTP proxy with name 'SERVE_CONTROLLER_ACTOR:SERVE_PROXY_ACTOR-dff7dc5b97b4a11facaed746f02448224aa0c1fb651988ba7197e949' on node 'dff7dc5b97b4a11facaed746f02448224aa0c1fb651988ba7197e949' listening on '127.0.0.1:8000'
```

File: doc/source/cluster/kubernetes/troubleshooting/rayservice-troubleshooting.md [hunk 2]
```diff
@@ -63,7 +63,7 @@ kubectl exec -it $HEAD_POD -- ray summary actors
 #     CLASS_NAME                          STATE_COUNTS
 # 0   ServeController                     ALIVE: 1
 # 1   ServeReplica:fruit_app_OrangeStand  ALIVE: 1
-# 2   HTTPProxyActor                      ALIVE: 3
+# 2   ProxyActor                      ALIVE: 3
```
Comment (zcin) [inline doc/source/cluster/kubernetes/troubleshooting/rayservice-troubleshooting.md:66]:
```suggestion
# 2   ProxyActor                          ALIVE: 3
```

**Review by zcin — APPROVED** [review]

File: doc/source/serve/advanced-guides/dev-workflow.md (hunk 1 above)
Comment (GeneDer) [reply doc/source/serve/advanced-guides/dev-workflow.md:55]: Oops thanks!

**Comment by zcin** [comment]
When I changed `http_proxies` -> `proxies` in schema.py, I forgot to modify the description: https://github.com/ray-project/ray/blob/master/python/ray/serve/schema.py#L902. Could you include a fix in this PR @GeneDer?

**Review by edoakes — APPROVED** [review]
nice! :)

**Review by architkulkarni — APPROVED** [review]
stamp

**Pull request merged**